ra/registration-authority_test.go

```diff
@@ -204,7 +225,48 @@ func initAuthorities(t *testing.T) (core.CertificateAuthority, *DummyValidationA
 	AuthzFinal.Expires = &exp
 	AuthzFinal.Challenges[0].Status = "valid"
 
-	return &ca, va, sa, &ra
+	return &ca, va, ssa, &ra, cleanUp
+}
+
+// This is an unfortunate bit of tech debt that is being taken on in
+// order to get the more important change of using MySQL/MariaDB in
+// all of our tests working without SQLite. We already had issues with
+// the RA here getting a real CertificateAuthority instead of a
+// CertificateAuthorityClient, so this is only marginally worse.
+func caDBImpl(t *testing.T) (core.CertificateAuthorityDatabase, func()) {
+	dbMap, err := sa.NewDbMap(dbConnStr)
+	if err != nil {
+		t.Fatalf("Could not construct dbMap: %s", err)
+	}
+
+	cadb, err := ca.NewCertificateAuthorityDatabaseImpl(dbMap)
+	if err != nil {
+		t.Fatalf("Could not construct CA DB: %s", err)
+	}
+
+	// We intentionally call CreateTablesIfNotExists twice before returning
+	// because of the weird insert inside it.
```

Comment:
Yeah, I will. For the record here: the serial id code is working by continuously getting and setting the same row that must exist before it can work. 

After we're on MySQL in all places, we could change this to use Flickr's id generating strategy that doesn't  require the row to exist ahead of time but that's for later.